### PR TITLE
Fix event name types

### PR DESCRIPTION
**Pull request by Codpoe**
Enhance TS types in TypedEmitter for better type prompts:

image

And in addition to the known event name, other event names can also work normally.

### CODE REVIEW

File: pnpm-lock.yaml
Comment (MatthewWid) [inline pnpm-lock.yaml:1]: Was this file accidentally committed? It doesn't seem relevant to this PR.

**Review by MatthewWid — CHANGES_REQUESTED** [review]
Thank you for contributing! Just a small fix-up needed.

File: pnpm-lock.yaml
Comment (Codpoe) [reply pnpm-lock.yaml:1]: Because the pnpm version is specified in pakcage.json `packageManager`, the lock file will automatically change when I install dependencies. I think this change may be necessary :)

Comment (MatthewWid) [reply pnpm-lock.yaml:1]: All good. I'll merge and make sure the lockfile is in a good state before releasing.
